scaletest/dashboard/run.go

```diff
@@ -0,0 +1,122 @@
+package dashboard
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"math/rand"
+	"time"
+
+	"golang.org/x/xerrors"
+
+	"cdr.dev/slog"
+	"github.com/coder/coder/codersdk"
+	"github.com/coder/coder/scaletest/harness"
+)
+
+type Runner struct {
+	client  *codersdk.Client
+	cfg     Config
+	metrics *Metrics
+}
+
+var (
+	_ harness.Runnable  = &Runner{}
+	_ harness.Cleanable = &Runner{}
+)
+
+func NewRunner(client *codersdk.Client, metrics *Metrics, cfg Config) *Runner {
+	client.Trace = cfg.Trace
+	return &Runner{
+		client:  client,
+		cfg:     cfg,
+		metrics: metrics,
+	}
+}
+
+func (r *Runner) Run(ctx context.Context, _ string, _ io.Writer) error {
+	me, err := r.client.User(ctx, codersdk.Me)
+	if err != nil {
+		return err
+	}
+	if len(me.OrganizationIDs) == 0 {
+		return xerrors.Errorf("user has no organizations")
+	}
+
+	c := &cache{}
+	if err := c.fill(ctx, r.client); err != nil {
+		return err
+	}
+
+	p := &params{
+		client: r.client,
+		me:     me,
+		c:      c,
+	}
+	rolls := make(chan int)
+	go func() {
+		t := time.NewTicker(r.randWait())
+		defer t.Stop()
+		for {
+			select {
+			case <-ctx.Done():
+				return
+			case <-t.C:
+				rolls <- rand.Intn(allActions.max() + 1) // nolint:gosec
```

Comment:
My main concern is that randomized tests may be hard to reproduce if every scenario is different. I'd rather model a few scenarios that cover all dashboard API calls. Feel free to challenge it though!

Reply:
> My main concern is that randomized tests may be hard to reproduce if every scenario is different.

We're not looking for 100% reproducibility here, I don't think. I was mainly looking to generate load that corresponds to a baseline number of requests per second. When performed over a period of time, you're going to end up with data points for all of the scenarios.

Reply:
I never even considered reproducibility for the scaletest. Is there a way to maybe log or use prom metrics to indicate how many of each "scenario" is run?

Not needed in this PR, but would be cool so if 1 scaletest just craps the bed, and we notice "Scenerio 5" was run an excessive amount of times. 

Idk how the metrics or logs are hooked up for this though

Reply:
100% adding prom metrics, we need those. 
Regarding reproucibility, we could use a seeded PRNG so we can seed a single test with the current time and be able to reproduce it if needed.